### ISSUE TITLE
SynchronousMachine4WindingsPowerSwing and most PowerGrids.Examples models compiled with NB fails because of missing C eqFunctions

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#12931]
Please check [PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing](https://libraries.openmodelica.org/branches/newInst-newBackend/PowerGrids/files/PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing.err), compiled with the NB. The linker fails with
```
PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0x3f): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_278'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0x54): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_277'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0x69): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_276'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0x7e): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_275'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0x93): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_274'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0xa8): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_273'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0xbd): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_272'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0xd2): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_271'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0xe7): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_270'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0xfc): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_269'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0x111): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_268'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0x126): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_267'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0x13b): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_266'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0x150): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_265'
/usr/bin/ld: PowerGrids_PowerGrids.Electrical.Test.SynchronousMachine4WindingsPowerSwing_09alg.c:(.text+0x165): undefined reference to `PowerGrids_PowerGrids_Electrical_Test_SynchronousMachine4WindingsPowerSwing_eqFunction_264'
```
@kabdelhak, @phannebohm something's wrong with function handling here.

This issue affects many models of [PowerGrids with the NB](https://libraries.openmodelica.org/branches/newInst-newBackend/PowerGrids/PowerGrids.html), so it's really high priority.

**Comment by casella** [comment]
Most likely a duplicate of #12242.

**Comment by casella** [comment]
Fixed by #12939.
